jctools-core/src/main/java/org/jctools/queues/SpscUnboundedArrayQueue.java

```diff
@@ -231,66 +173,17 @@ private E newBufferPeek(E[] nextBuffer, final long index, final long mask) {
         return (E) lvElement(nextBuffer, offsetInNew);// LoadLoad
     }
 
-    @Override
-    public final int size() {
-        /*
-         * It is possible for a thread to be interrupted or reschedule between the read of the producer and
-         * consumer indices, therefore protection is required to ensure size is within valid range. In the
-         * event of concurrent polls/offers to this method the size is OVER estimated as we read consumer
-         * index BEFORE the producer index.
-         */
-        long after = lvConsumerIndex();
-        while (true) {
-            final long before = after;
-            final long currentProducerIndex = lvProducerIndex();
-            after = lvConsumerIndex();
-            if (before == after) {
-                return (int) (currentProducerIndex - after);
-            }
-        }
-    }
-
-    private void adjustLookAheadStep(int capacity) {
-        producerLookAheadStep = Math.min(capacity / 4, MAX_LOOK_AHEAD_STEP);
-    }
-
-    private long lvProducerIndex() {
-        return UNSAFE.getLongVolatile(this, P_INDEX_OFFSET);
-    }
-
-    private long lvConsumerIndex() {
-        return UNSAFE.getLongVolatile(this, C_INDEX_OFFSET);
-    }
-
-    private void soProducerIndex(long v) {
-        UNSAFE.putOrderedLong(this, P_INDEX_OFFSET, v);
-    }
-
-    private void soConsumerIndex(long v) {
-        UNSAFE.putOrderedLong(this, C_INDEX_OFFSET, v);
-    }
-
     private static long calcWrappedOffset(long index, long mask) {
```

Comment:
Perhaps this method should be replaced with the CircularArrayCalculator utility method